crates/tabby-common/src/registry.rs

```diff
@@ -40,15 +40,27 @@ fn models_json_file(registry: &str) -> PathBuf {
 }
 
 async fn load_remote_registry(registry: &str) -> Result<Vec<ModelInfo>> {
-    let model_info = reqwest::get(format!(
-        "https://raw.githubusercontent.com/{}/registry-tabby/main/models.json",
-        registry
-    ))
-    .await
-    .context("Failed to download")?
-    .json()
-    .await
-    .context("Failed to get JSON")?;
+    // Create a client with custom timeout.
+    // There have been instances where requests did not adhere to the default timeout,
+    // resulting in requests hanging indefinitely.
+    // Therefore, it's necessary to specify a custom timeout.
```

Comment:
```suggestion
    // Create an HTTP client with a custom timeout.
    // This is necessary because the default timeout settings can sometimes cause requests to hang indefinitely.
    // To prevent such issues, we specify a custom timeout duration.
```